### PR TITLE
Exporter code simplification

### PR DESCRIPTION
**Pull request by trask**
No need to remove semantic attributes during mapping, since not capturing any of them anyways.

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
     private static final Logger logger = LoggerFactory.getLogger(TraceIdBasedSampler.class);
 
-    private static final AttributeKey<Double> AI_SAMPLING_PERCENTAGE = AttributeKey.doubleKey("ai.internal.sampling.percentage");
+    private static final AttributeKey<Double> AI_SAMPLING_PERCENTAGE = AttributeKey.doubleKey("applicationinsights.internal.sampling_percentage");
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java:23]: can we keep "sampling.percentage"?

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 2]
```diff
@@ -72,12 +70,16 @@
 
     private static final Joiner JOINER = Joiner.on(", ");
 
-    private static final AttributeKey<Double> AI_SAMPLING_PERCENTAGE = AttributeKey.doubleKey("ai.internal.sampling.percentage");
+    private static final AttributeKey<Double> AI_SAMPLING_PERCENTAGE_KEY = AttributeKey.doubleKey("applicationinsights.internal.sampling_percentage");
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:73]: "sampling.percentage.key"?

Comment (heyams) [reply agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:73]: note that AI_SAMPLING_PERCENTAGE and AI_SAMPLING_PERCENTAGE_KEY have the same value.

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 3]
```diff
@@ -72,12 +70,16 @@
...
 
-    private static final AttributeKey<Boolean> AI_INTERNAL_LOG = AttributeKey.booleanKey("ai.internal.log");
+    private static final AttributeKey<Boolean> AI_LOG_KEY = AttributeKey.booleanKey("applicationinsights.internal.log");
 
-    private static final AttributeKey<String> SPAN_SOURCE_ATTRIBUTE_NAME = AttributeKey.stringKey(AiAppId.SPAN_SOURCE_ATTRIBUTE_NAME);
-    private static final AttributeKey<String> SPAN_TARGET_ATTRIBUTE_NAME = AttributeKey.stringKey(AiAppId.SPAN_TARGET_ATTRIBUTE_NAME);
+    private static final AttributeKey<String> AI_SPAN_SOURCE_KEY = AttributeKey.stringKey(AiAppId.SPAN_SOURCE_ATTRIBUTE_NAME);
+    private static final AttributeKey<String> AI_SPAN_TARGET_KEY = AttributeKey.stringKey(AiAppId.SPAN_TARGET_ATTRIBUTE_NAME);
+
+    private static final AttributeKey<String> AI_LOG_LEVEL_KEY = AttributeKey.stringKey("applicationinsights.internal.log_level");
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:80]: "log.level"?

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 4]
```diff
@@ -72,12 +70,16 @@
...
+    private static final AttributeKey<String> AI_LOGGER_NAME_KEY = AttributeKey.stringKey("applicationinsights.internal.logger_name");
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:81]: logger.name?

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 5]
```diff
@@ -72,12 +70,16 @@
...
+    private static final AttributeKey<String> AI_LOG_ERROR_STACK_KEY = AttributeKey.stringKey("applicationinsights.internal.log_error_stack");
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:82]: log.error.stack?

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 6]
```diff
@@ -116,58 +118,61 @@ private void export(SpanData span) {
...
 
-    private void exportRequest(SpanData span, Map<AttributeKey<?>, Object> attributes) {
+    private void exportRequest(SpanData span) {
 
         RequestTelemetry telemetry = new RequestTelemetry();
 
         String source = null;
-        String sourceAppId = removeAttributeString(attributes, SPAN_SOURCE_ATTRIBUTE_NAME);
+        ReadableAttributes attributes = span.getAttributes();
+        String sourceAppId = attributes.get(AI_SPAN_SOURCE_KEY);
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:151]: can attributes become null?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java (hunk 1 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java:23]: Look like AI_SAMPLING_PERCENTAGE is no longer used.

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 7]
```diff
@@ -243,46 +242,43 @@ private void exportRemoteDependency(SpanData span, Map<AttributeKey<?>, Object>
...
-        String loggerName = removeAttributeString(attributes, LOGGER_LOGGER_NAME);
-        String errorStack = removeAttributeString(attributes, LOGGER_ERROR_STACK);
-        Double samplingPercentage = removeAiSamplingPercentage(attributes);
+        ReadableAttributes attributes = span.getAttributes();
+        String level = attributes.get(AI_LOG_LEVEL_KEY);
+        String loggerName = attributes.get(AI_LOGGER_NAME_KEY);
+        String errorStack = attributes.get(AI_LOG_ERROR_STACK_KEY);
+        Double samplingPercentage = attributes.get(AI_SAMPLING_PERCENTAGE_KEY);
         if (errorStack == null) {
             trackTrace(message, span.getStartEpochNanos(), level, loggerName, span.getTraceId(),
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:283]: can we simply pass span to trackTrace(Span span)?  it has 8 input parameters.

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 8]
```diff
@@ -320,22 +316,23 @@ private void trackEvents(SpanData span, Double samplingPercentage) {
...
         }
 
-        setProperties(telemetry.getProperties(), level, loggerName, attributes);
+        setLoggerProperties(telemetry.getProperties(), level, loggerName);
+        setExtraAttributes(telemetry.getProperties(), attributes);
         telemetry.setTimestamp(new Date(NANOSECONDS.toMillis(timeEpochNanos)));
         track(telemetry, samplingPercentage);
     }
 
     private void trackTraceAsException(String message, long timeEpochNanos, String level, String loggerName,
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:333]: same here: trackTraceAsException(Span span)?

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java [hunk 9]
```diff
@@ -572,76 +562,30 @@ private static void addLinks(Map<String, String> properties, List<Link> links) {
...
+    private static final Set<String> STANDARD_ATTRIBUTE_PREFIXES =
+            ImmutableSet.of("http", "db", "message", "messaging", "rpc", "enduser", "net", "peer", "exception", "thread", "faas");
 
-    private static void addExtraAttributes(Map<String, String> properties, Attributes attributes) {
+    private static void setExtraAttributes(Map<String, String> properties, ReadableAttributes attributes) {
         attributes.forEach(new AttributeConsumer() {
             @Override
             public <T> void accept(AttributeKey<T> key, T value) {
+                String stringKey = key.getKey();
+                if (stringKey.startsWith("applicationinsights.internal.")) {
```
Comment (heyams) [inline agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:573]: do you mean negation? !startsWith.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/TraceIdBasedSampler.java:23]: this is otel convention, dots to separate namespaces, and underscores to separate words

`AI_SAMPLING_PERCENTAGE` is still used below

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java (hunk 6 above)
Comment (trask) [reply agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:151]: they will be null if they were never set

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java (hunk 2 above)
Comment (trask) [reply agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java]: good point, I consolidated them

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java (hunk 7 above)
Comment (trask) [reply agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java]: oh yes! that's much better

File: agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java (hunk 9 above)
Comment (trask) [reply agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java:577]: it's skipping all internal attributes, we don't want to send them to breeze

**Comment by heyams** [comment]
Might need to double check unit tests for trackTrace and trackTraceException for the signature changes.

**Review by heyams — APPROVED** [review]

**Pull request merged**